packages/plugins/documentation/server/utils/builders/build-api-endpoint-path.js

```diff
@@ -0,0 +1,132 @@
+    .map(token => {
+      if (_.isObject(token)) {
+        return token.prefix + '{' + token.name + '}';
+      }
+
+      return token;
+    })
+    .join('');
+};
+
+/**
+ * @description Builds the required object for a path parameter
+ *
+ * @param {string} routePath - The route's path property
+ *
+ * @returns Swagger path params object
+ */
+const getPathParams = routePath => {
+  return pathToRegexp
+    .parse(routePath)
+    .filter(token => _.isObject(token))
+    .map(param => {
+      return {
+        name: param.name,
+        in: 'path',
+        description: '',
+        deprecated: false,
+        required: true,
+        schema: { type: 'string' },
+      };
+    });
+};
+
+/**
+ *
+ * @param {array} routes - The routes for a given api or plugin
+ * @param {object} attributes - The attributes for a given api or plugin
+ * @param {string} tag - A descriptor for OpenAPI
+ *
+ * @returns object of OpenAPI paths for each route
+ */
+const getPaths = (routes, attributes, tag) => {
+  const paths = routes.reduce(
+    (acc, route) => {
+      // TODO: Find a more reliable way to determine list of entities vs a single entity
+      const isListOfEntities = route.handler.split('.').pop() === 'find';
+      const hasPathParams = route.path.includes('/:');
+      const methodVerb = route.method.toLowerCase();
+      const routePath = hasPathParams ? parsePathWithVariables(route.path) : route.path;
+
+      const { responses } = buildApiResponses(attributes, route, isListOfEntities);
+      _.set(acc.paths, `${routePath}.${methodVerb}.responses`, responses);
+      _.set(acc.paths, `${routePath}.${methodVerb}.tags`, [_.upperFirst(tag)]);
+
+      if (isListOfEntities) {
+        _.set(acc.paths, `${routePath}.${methodVerb}.parameters`, queryParams);
+      }
+
+      if (hasPathParams) {
+        const pathParams = getPathParams(route.path);
+        _.set(acc.paths, `${routePath}.${methodVerb}.parameters`, pathParams);
+      }
+
+      if (methodVerb === 'post' || methodVerb === 'put') {
+        const { requestBody } = buildApiRequests(attributes, route);
+
+        _.set(acc.paths, `${routePath}.${methodVerb}.requestBody`, requestBody);
+      }
+
+      return acc;
+    },
+    { paths: {} }
+  );
+
+  return paths;
+};
+
+/**
+ * @description - Builds the Swagger paths object for each api
+ *
+ * @param {object} api - Information about the current api
+ * @property {string} api.name - The name of the api
+ * @property {string} api.getter - The getter for the api (api | plugin)
+ * @property {array} api.ctNames - The name of all contentTypes found on the api
+ *
+ * @returns
+ */
+module.exports = api => {
+  if (!api.ctNames.length && api.getter === 'plugin') {
+    // Set arbitrary attributes
+    const attributes = { foo: { type: 'string' } };
+    // No contenType because it uses admin routes?
+    const routes = strapi.plugin(api.name).routes['admin'].routes;
+
+    return getPaths(routes, attributes, api.name);
+  }
```

Comment:
ℹ️ This needs to be fixed.  It is a quick fix because the 'email' plugin  doesn't have any contentTypes and to match the v3 plugin it needs to use admin routes.